portfolio/src/main/webapp/script.js

```diff
@@ -90,11 +91,35 @@ function createCommentElement(comment) {
   const dateReadable = (new Date(comment.timestamp)).toDateString();
   nameElement.text(`${comment.name} at ${dateReadable}`);
 
+  const deleteBtn = $("<button></button>");
+  deleteBtn.text("Delete");
+  deleteBtn.on("click", async () => {
+    await deleteComment(id);
+  })
+
   node.append(contentElement);
   node.append(nameElement);
+  node.append(deleteBtn);
   return node;
 }
 
+/**
+ * Delete a comment from the Datastore given the id. Alerts the user if something goes wrong and comment cannot be deleted.
+ */
+async function deleteComment(id) {
+  const params = new URLSearchParams();
+  params.append('id', id);
+  const resp = await fetch('/delete-data', {method: 'POST', body: params});
+  if (!resp.ok) {
+    alert("AH YIKES! Cannot delete comment");
```

Comment:
Optional: alert is generally not a good user experience :) 

Consider using something like toast/snackbar (https://material.io/develop/web/components/snackbars/) to show the error message to the users.

Reply:
ohh I'll look into this -- thank you!